server/protocols/webadmin/webadmin-cassandra-data/src/test/java/org/apache/james/webadmin/routes/CassandraMappingsRoutesTest.java

```diff
@@ -0,0 +1,222 @@
+        mappingsSourcesMigration = new MappingsSourcesMigration(cassandraRecipientRewriteTableDAO, cassandraMappingsSourcesDAO);
+
+        CassandraMappingsService cassandraMappingsService = new CassandraMappingsService(mappingsSourcesMigration, cassandraMappingsSourcesDAO);
+
+        JsonTransformer jsonTransformer = new JsonTransformer();
+        taskManager = new MemoryTaskManager();
+        webAdminServer = WebAdminUtils.createWebAdminServer(
+            new DefaultMetricFactory(),
+            new CassandraMappingsRoutes(cassandraMappingsService, taskManager, jsonTransformer),
+            new TasksRoutes(taskManager, jsonTransformer));
+
+        webAdminServer.configure(NO_CONFIGURATION);
+        webAdminServer.await();
+
+        RestAssured.requestSpecification = WebAdminUtils.buildRequestSpecification(webAdminServer)
+            .setBasePath(CassandraMappingsRoutes.ROOT_PATH)
+            .build();
+    }
+
+    @After
+    public void tearDown() {
+        webAdminServer.destroy();
+        taskManager.stop();
+        cassandra.clearTables();
+    }
+
+    @AfterClass
+    public static void tearDownClass() {
+        cassandra.closeCluster();
+    }
+
+    @Test
+    public void postMappingsActionWithSolvedInconsistenciesQueryParamShouldCreateATask() {
+        given()
+            .queryParam("action", MAPPINGS_ACTION)
+        .when()
+            .post()
+        .then()
+            .statusCode(HttpStatus.CREATED_201)
+            .header("Location", is(notNullValue()))
+            .body("taskId", is(notNullValue()));
+    }
+
+    @Test
+    public void postMappingsActionWithSolvedInconsistenciesQueryParamShouldHaveSuccessfulCompletedTask() {
+        String taskId = with()
+            .queryParam("action", MAPPINGS_ACTION)
+            .post()
+            .jsonPath()
+            .get("taskId");
+
+        given()
+            .basePath(TasksRoutes.BASE)
+        .when()
+            .get(taskId + "/await")
+        .then()
+            .body("status", is("completed"))
+            .body("taskId", is(notNullValue()))
+            .body("type", is(CassandraMappingsSolveInconsistenciesTask.TYPE))
+            .body("startedDate", is(notNullValue()))
+            .body("submitDate", is(notNullValue()))
+            .body("completedDate", is(notNullValue()));
+    }
+
+    @Test
+    public void postMappingsActionShouldRejectInvalidActions() {
+        given()
+            .queryParam("action", "invalid-action")
+        .when()
+            .post()
+        .then()
+            .statusCode(HttpStatus.BAD_REQUEST_400)
+            .body("statusCode", is(400))
+            .body("type", is(ErrorResponder.ErrorType.INVALID_ARGUMENT.getType()))
+            .body("message", is("Invalid arguments supplied in the user request"))
+            .body("details", is("No enum constant org.apache.james.webadmin.dto.ActionMappings.invalid-action"));
```

Comment:
is that okay to leak james internal information(`...org.apache.james.webadmin.dto.ActionMappings...`) to outside?
in the other hand, this response message with me is not clear enough to the clients.

Reply:
Hmmm... It's being generated in a generic way by the new service.exception introduced in `WebAdminServer`. How would you suggest to improve the error message in that case?

Reply:
By generating  a better exception when parsing the action in the first place. I can help you do this tomorrow.